src/pricecypher/rest.py

```diff
@@ -9,6 +11,68 @@
 
 UNKNOWN_ERROR = 'pricecypher.sdk.internal.unknown'
 
+# Mapping from (wildcard) domains to IP addresses to resolve manually.
+dns_cache = {}
+# Keep track of original getaddrinfo function to use for domain names not in our custom cache.
+prv_getaddrinfo = socket.getaddrinfo
+
+
+def is_ipv4(s):
+    """ Check whether the given address is an IPv4 address. """
+    return ':' not in s
+
+
+def add_custom_dns(domain, port, ip):
+    """
+    Add a custom DNS record to our local cache, such that requests for the given domain and port are resolved to the
+    given ip address.
+    ."""
+    # See docs for tuple format: https://docs.python.org/2/library/socket.html#socket.getaddrinfo
+    if is_ipv4(ip):
+        value = (socket.AddressFamily.AF_INET, 1, 6, '', (ip, port))
+    else:  # ipv6
+        value = (socket.AddressFamily.AF_INET6, 1, 6, '', (ip, port, 0, 0))
+
+    dns_cache[domain] = [value]
+
+
+def new_getaddrinfo(*args):
+    """
+    Define a patched function that will replace the original `socket.getaddrinfo` function.
+    This patched function will first check if the given host or its wildcard host is included in our local cache.
+    If so, it resolves the address to the custom IP in our cache. Otherwise, the original getaddrinfo is used instead.
+    """
+    # Find host in given arguments and split it into parts.
+    host = args[0]
+    host_parts = host.split('.')
+
+    if len(host_parts) < 1:
+        return prv_getaddrinfo(*args)
+
+    # Replace first part of domain with * such that we can also look for the wildcard host in our local cache.
+    host_parts[0] = '*'
```

Comment:
Does it matter if the domain is for example `test.customer.pricecypher.com`? Because it will try to find `*.customer.pricecypher.com` which may not exist, but `*.pricecypher.com` could exist, which it would not find, but still applies. Then the resolution would not be as expected right?

Reply:
The resolution logic does not need to be that advanced.